packages/api/models/granules.js

```diff
@@ -446,16 +420,72 @@ class Granule extends Manager {
       } else {
         const operation = '=';
         attributeValues[`:${key}`] = value;
-        expression = `#${field} ${operation} :${key}`;
+        if (!isQuery && field === 'granuleId') {
+          expression = `contains(#${field}, :${key})`;
+        } else {
+          expression = `#${field} ${operation} :${key}`;
+        }
       }
 
-      if (field === 'granuleId') {
+      if (isQuery && field === 'granuleId') {
         keyConditionArray.push(expression);
       } else {
         filterArray.push(expression);
       }
     });
 
+    return {
+      attributeNames,
+      attributeValues,
+      filterArray,
+      filterExpression: (filterArray.length > 0) ? filterArray.join(' AND ') : undefined,
+      keyConditionArray,
+      keyExpression: keyConditionArray.join(' AND '),
+    };
+  }
+
+  /**
+   * return the queue of the granules for a given collection,
+   * the items are ordered by granuleId
+   *
+   * @param {string} collectionId - collection id
+   * @param {string} status - granule status, optional
+   * @returns {Array<Object>} the granules' queue for a given collection
+   */
+  getGranulesForCollection(collectionId, status) {
+    const fields = ['granuleId', 'collectionId', 'files', 'published', 'createdAt'];
+    const searchParams = status ? { status } : {};
+    return this.searchGranulesForCollection(collectionId, searchParams, fields);
+  }
+
+  /**
+   * return the queue of the granules for a given collection and search params,
+   * the items are ordered by granuleId
+   *
+   * @param {string} collectionId - collection id
+   * @param {Object} searchParams - optional, search parameters
+   * @param {Array<string>} fields - optional, fields to return
+   * @returns {Array<Object>} the granules' queue for a given collection
+   */
+  searchGranulesForCollection(collectionId, searchParams = {}, fields = []) {
+    // Key Condition Expression does not support IN operators
+    if (searchParams.granuleId && !isString(searchParams.granuleId)) {
+      throw new CumulusModelError('Could not search granule records for collection, granuleId is not string');
+    }
+    // Filter Expression can't contain key attributes
+    if (searchParams.collectionId) {
+      throw new CumulusModelError('Could not search granule records for collection, do not specify collectionId in searchParams');
+    }
+
+    const {
+      attributeNames,
+      attributeValues,
+      filterExpression,
+      keyConditionArray,
```

Comment:
use `keyExpression` instead?

Reply:
Stopped `keyExpression` from being returned in `getDynamoDbSearchParams`. Want to use `keyConditionArray` here because we end up adding the collectionId to that array later in this function

`keyConditionArray.push('#collectionId = :collectionId');`